apps/web/server/lib/ssr.ts

```diff
@@ -16,20 +16,22 @@ import { appRouter } from "@calcom/trpc/server/routers/_app";
 export async function ssrInit(context: GetServerSidePropsContext) {
   const ctx = await createContext(context);
   const locale = getLocaleFromHeaders(context.req);
-  const i18n = await serverSideTranslations(getLocaleFromHeaders(context.req), ["common", "vital"]);
+  const i18n = await serverSideTranslations(locale, ["common", "vital"]);
 
   const ssr = createProxySSGHelpers({
     router: appRouter,
     transformer: superjson,
     ctx: { ...ctx, locale, i18n },
   });
 
-  // always preload "viewer.public.i18n"
-  await ssr.viewer.public.i18n.fetch();
-  // So feature flags are available on first render
-  await ssr.viewer.features.map.prefetch();
-  // Provides a better UX to the users who have already upgraded.
-  await ssr.viewer.teams.hasTeamPlan.prefetch();
+  await Promise.all([
```

Comment:
we are trying to reduce the usage of Promise.all to prevent reaching the limit of database connections. not sure if we should do this. what do you think @emrysal  ?

Reply:
Oh, didn't know that! Well, if so, then yes, it will make 1 additional connection here.

Reply:
https://www.youtube.com/watch?v=f2Z1v3cqgDI

Reply:
> https://www.youtube.com/watch?v=f2Z1v3cqgDI

Agreed that it can be overused, but in this case we just need to run 3 queries with no side effects and no error handling, and since this method runs basically on every page, it makes sense to speed it up a little bit.

If the goal is to limit db connections, then I agree that it's not a good idea to parallelize things, but at the same time I think it would be better to solve this problem with a db bouncer instead of relying on waterfall code?